src/NuGet.Core/NuGet.Protocol/Converters/VersionRangeConverter.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Newtonsoft.Json;
+using NuGet.Versioning;
+using System;
+
+namespace NuGet.Protocol
+{
+    /// <summary>
+    /// A VersionRange JSON converter.
+    /// </summary>
+    public class VersionRangeConverter : JsonConverter
```

Comment:
Woah this has never been done before? How does deserializing registration pages work? These contain dependency ranges AFAIK.

Reply:
There's a `PackageDependencyGroupConverter` for that which targets the full `PackageDependencyGroup` type.